openforcefield/tests/test_examples.py

```diff
@@ -97,17 +97,42 @@ def find_readme_examples():
     return re.findall('```python(.*?)```', readme_content, flags=re.DOTALL)
 
 
+def find_readme_links():
+    """Yield all the links in the main README.md file.
+
+    Returns
+    -------
+    readme_examples : List[str]
+        The list of links included in the README.md file.
+    """
+    readme_file_path = os.path.join(ROOT_DIR_PATH, 'README.md')
+    with open(readme_file_path, 'r') as f:
+        readme_content = f.read()
+    return re.findall('http[s]?://(?:[0-9a-zA-Z]|[-/.%:])+', readme_content)
+
+
 #======================================================================
 # TESTS
 #======================================================================
 
 @pytest.mark.parametrize('readme_example_str', find_readme_examples())
-def test_readme(readme_example_str):
+def test_readme_examples(readme_example_str):
     """Test the example"""
     run_script_str(readme_example_str)
 
 
+@pytest.mark.parametrize('readme_link', find_readme_links())
+def test_readme_links(readme_link):
+    """Test the example"""
+    from urllib.request import Request, urlopen
+    headers = {'User-Agent':'Mozilla/5.0',
```

Comment:
This `headers` section seems a little arcane, and a google search didn't make it immediately clear what this is doing. Could you include a comment on what this does, in case we need to fix it in the future?

Reply:
Will do!